Yesod/Auth/OAuth2/Github.hs

```diff
@@ -119,3 +119,6 @@ toCreds user userMails token = Creds
 
     maybePublicEmail Nothing = []
     maybePublicEmail (Just e) = [("public_email", e)]
+
+    maybeLocation Nothing = []
+    maybeLocation (Just location) = [("location", location)]
```

Comment:
Rule of three says we should think about extracting something now:

```hs
++ maybeKey "name" (githubUserName user)
++ maybeKey "public_email" (githubUserPublicEmail user)
++ maybeKey "location" (githubUserLocation user)

maybeKey :: Text -> Maybe Text -> [(Text, Text)]
maybeKey k (Just x) = [(k, v)]
maybeKey _ Nothing = []

-- Or maybe something fancier, with TupleSections?
maybeKey k = maybeToList . fmap (k,)
```

WDYT? (I'm not in love with the "maybe-key" name, FWIW).